inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+            data.append_attribute("offset").set_value(attr.offset);
+            data.append_attribute("size").set_value(attr.size);
+        }
+
+        int port_id = 0;
+        // <layers/input>
+        if (node->get_input_size() > 0) {
+            pugi::xml_node input = layer.append_child("input");
+            for (auto i : node->inputs()) {
+                NGRAPH_CHECK(i.get_partial_shape().is_static(),
+                             "Dynamic shape of ", node->get_name(), " (",
+                             node->get_type_name(), ") ", "is not supported.");
+
+                pugi::xml_node port = input.append_child("port");
+                port.append_attribute("id").set_value(port_id++);
+                for (auto d : i.get_shape()) {
+                    pugi::xml_node dim = port.append_child("dim");
+                    dim.append_child(pugi::xml_node_type::node_pcdata)
+                        .set_value(std::to_string(d).c_str());
+                }
+            }
+        }
+        // <layers/output>
+        if ((node->get_output_size() > 0) &&
+            !dynamic_cast<ngraph::op::Result*>(node)) {
+            pugi::xml_node output = layer.append_child("output");
+            for (auto o : node->outputs()) {
+                NGRAPH_CHECK(o.get_partial_shape().is_static(),
+                             "Dynamic shape of ", node->get_name(), " (",
+                             node->get_type_name(), ") ", "is not supported.");
+
+                pugi::xml_node port = output.append_child("port");
+                port.append_attribute("id").set_value(port_id++);
+                port.append_attribute("precision")
+                    .set_value(get_output_precision_name(o).c_str());
+                for (auto d : o.get_shape()) {
+                    pugi::xml_node dim = port.append_child("dim");
+                    dim.append_child(pugi::xml_node_type::node_pcdata)
+                        .set_value(std::to_string(d).c_str());
+                }
+            }
+        }
+    }
+    // <edges>
+    const std::vector<Edge> edge_mapping = create_edge_mapping(layer_ids, f);
+    pugi::xml_node edges = netXml.append_child("edges");
+    for (auto e : edge_mapping) {
+        pugi::xml_node edge = edges.append_child("edge");
+        edge.append_attribute("from-layer").set_value(e.from_layer);
+        edge.append_attribute("from-port").set_value(e.from_port);
+        edge.append_attribute("to-layer").set_value(e.to_layer);
+        edge.append_attribute("to-port").set_value(e.to_port);
+    }
+}
+
+}  // namespace
+
+// ! [function_pass:serialize_cpp]
+// serialize.cpp
+bool pass::Serialize::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    // prepare data
+    pugi::xml_document xml_doc;
+    std::vector<uint8_t> constants;
+    switch (m_version) {
+    case Version::IR_V10:
+        ngfunction_2_irv10(xml_doc, constants, *f);
+        break;
+    default:
+        NGRAPH_UNREACHABLE("");
```

Comment:
nit: Please add an error message.

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/48104070eae5a01692536a45e7591760ac7e06d6)